### PR TITLE
Bypass the restore in GRPC/k8s mode if the backup doesn't exist

### PR DESCRIPTION
**Pull request by adejanovski**
Fixes https://github.com/k8ssandra/k8ssandra/issues/461

### CODE REVIEW

**Comment by adejanovski** [comment]
@jsanda, I've tested it locally using this image: 
```
image:
    repository: docker.io/adejanovski/medusa
    tag: restore-fix
```


I'll create an integration test for this bug in the k8ssandra project.

File: medusa/service/grpc/restore.py [hunk 1]
```diff
@@ -65,9 +66,17 @@ def configure_console_logging(config):
...
+backup_found = False
+# Checking if the backup exists for the node we're restoring.
+# Skipping restore if it doesn't exist.
+for cluster_backup in cluster_backups:
+    if cluster_backup.name == backup_name:
+        backup_found = True
+        logging.info("Starting restore of backup {}".format(backup_name))
+        medusa.restore_node.restore_node(config, tmp_dir, backup_name, in_place, keep_auth,
+                                         seeds, verify, keyspaces, tables, use_sstableloader)
+        logging.info("Finished restore of backup {}".format(backup_name))
```
Comment (jsanda) [inline medusa/service/grpc/restore.py:79]: I know that backup names are unique which means we should only get into this `if` block once. Would it good to add a `break` statement at the end after line 79?

**Review by jsanda — COMMENTED** [review]
Thanks!

File: medusa/service/grpc/restore.py (hunk 1 above)
Comment (adejanovski) [reply medusa/service/grpc/restore.py:79]: oooh yeah, most definitely 👍

Comment (adejanovski) [reply medusa/service/grpc/restore.py:79]: done

**Review by jsanda — APPROVED** [review]

**Pull request merged**